JoinColony/colonyCDapp
# Feat: Cancel streams using permissions

[See block-ingestor counterpart](https://github.com/JoinColony/block-ingestor/pull/217)

## Description

Users should be able to cancel streaming payments.

This PR implements a new saga and a block-ingestor handler.

## Testing

1. Install the streaming payments extension
2. Create a new streaming payment using the temporary UI on the dashboard
3. Run this query and note the end time of the streaming payment

```
query MyQuery {
  listStreamingPayments {
    items {
      endTime
      startTime
      nativeId
    }
  }
}
```

4. Using the temporary dashboard UI, cancel the streaming payment (ids start at 1)
5. Run the query again and notice the endTime change
6. Verify a new colony action was created by running this query. The latest action should have a streamingPaymentId.

```
query MyQuery {
  listColonyActions {
    items {
      streamingPaymentId
      id
      streamingPayment {
        id
        endTime
        createdAt
      }
    }
  }
}
```

## Diffs

**New stuff** ✨

* Cancel streaming payment saga

**Changes** 🏗

* Colony action now accepts streaming payment ID

Resolves #2215
